lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -372,6 +372,25 @@ def _lsblk_uuid(self):
 
         return uuids
 
+    def _udevadm_uuid(self, device):
+        # fallback for versions of lsblk <= 2.23 that don't have --paths, see _run_lsblk() above
+        uuid = 'N/A'
+
+        udevadm_path = self.module.get_bin_path('udevadm')
+        if not udevadm_path:
+            return uuid
+
+        cmd = [udevadm_path, 'info', '--query', 'property', '--name=%s' % device]
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            return uuid
+
+        m = re.search('ID_FS_UUID=(.*)\n', out)
```

Comment:
I would like to see an example output of 'udevadm info --query property --name  $DEVICE' and a test case that uses it to verify the regex here. It may be easier to test if the parsing bit is split into a separate method.

(granted, that is a very simple regex at the moment...)

Reply:
:+1: unit test added